core/src/wsv.rs

```diff
@@ -615,23 +645,34 @@ impl WorldStateView {
             // TODO: If we cannot apply the block, it is preferred to
             // signal failure and have the end user figure out what's
             // wrong.
-            self.apply(block)
+            self.apply(&block)
                 .expect("World state View failed to apply.");
         }
     }
 
-    /// Hash of nth back block
-    pub fn nth_back_block_hash(&self, n: usize) -> HashOf<VersionedCommittedBlock> {
-        self.blocks
-            .nth_back(n)
-            .map_or(Hash::zeroed().typed(), |block| block.value().hash())
+    /// Hash of latest block
+    pub fn latest_block_hash(&self) -> HashOf<VersionedCommittedBlock> {
+        self.block_hashes
+            .borrow()
+            .iter()
+            .last()
+            .map_or(Hash::zeroed().typed(), |hash| *hash)
     }
 
     /// View change index of nth back block
-    pub fn nth_back_block_view_change_index(&self, n: usize) -> u64 {
-        self.blocks
-            .nth_back(n)
-            .map_or(0, |block| block.value().header().view_change_index)
+    pub fn latest_block_view_change_index(&self) -> u64 {
+        self.kura
+            .get_block_by_height(self.height())
+            .map_or(0, |block| block.header().view_change_index)
+    }
+
+    /// Hash of latest block
+    pub fn previous_block_hash(&self) -> HashOf<VersionedCommittedBlock> {
+        self.block_hashes
+            .borrow()
+            .iter()
+            .last()
+            .map_or(Hash::zeroed().typed(), |hash| *hash)
```

Comment:
We don't return previous to latest block hash from this method. 

I would also suggest to preserve `nth_back` method (especially since iterator of `Vec` is `DoubleEndedIterator` and provide us this method for free).

Reply:
Very foolish of me for not implementing the function properly. I have fixed it now.

I prefer keeping these helper functions more specific. It expresses more intent.